lib/pengine/rules.c

```diff
@@ -58,7 +57,6 @@ pe_test_rule(xmlNode *rule, GHashTable *node_hash, enum rsc_role_e role,
 {
     pe_rule_eval_data_t rule_data = {
         .node_hash = node_hash,
-        .role = role,
```

Comment:
Deprecate the role arguments in `pe_test_rule()` and `pe_test_expression()`

Reply:
Updated pe_test_expression() doxygen, not going to add new doxygen for pe_test_rule() b/c I plan to deprecate the entire function